rigging/dbus_connection.py

```diff
@@ -0,0 +1,105 @@
+import dbus
+import dbus.service
+import dbus.mainloop.glib
+from gi.repository import GLib
+from rigging.exceptions import (DBusServiceExistsError, 
+                            DBusServiceDoesntExistError,
+                            DBusMethodDoesntExistError)
+
+class RigDBusMessage:
+    result = None
+    success = None
+
+    def __init__(self, result=None, success=None):
+        self.result = result
+        self.success = success
+
+class RigDBusCommand:
+    name = None
+
+class RigDBusCommandDestroy(RigDBusCommand):
+    name = "destroy"
```

Comment:
Rather than a separate class for each command, would it make sense to instead build the `RigDbusCommand` instance for each communication? This could allow us to more easily extend some kind of auth mechanism down the line.

E.G.

```python
conn = RigDbusConnection('myrig')
conn.destroy_rig()

[...]

def destroy_rig():
    self._communicate(RigDbusCommand("destroy", self.get_some_auth_bits())
[...]
```

Reply:
Sure, that makes sense and makes things easier. I'll change this

Reply:
Done 873ae2031dfb7012490dd809617f158128d70969